### ISSUE TITLE
Partition Array Into K Equal Sum Subsets

### ISSUE THREAD

**Issue by codewithcs** [imnishant/Geeks-For-Geeks#7]
Hi, 

Do you know what vis[temp-1][i] mean in your solution to Partition array into k equal sum subsets problem ?

**Comment by imnishant** [comment]
Hi Shikhar,

Firstly I have used 2d boolean matrix '**vis**' to apply Dynamic Programming concept. If you will not use it, then it will take more time to give output. Now its dimensions are **[sum / k][n]** where 'sum / k' will be the sum of elements of the individual array after partition (It will be the same for all individual array after partition). Since index can range from (0 to (sum / k) - 1) so I have used (temp - 1) where 'temp' is intermediate sum at any time. 

Now to avoid solving the same subproblems I have used it in the code. For example, let us suppose the value of 'temp' is 5 and the value of 'targetSum' is 10 and we are at position 'i = 7' in the array. Now if the element on position 'i = 7' does not lead to the 'targetSum' for the current value of 'temp', I am marking **vis[temp - 1][i] = true**, it means if in future when we will be finding other subsets and we land up to the same subproblem which is 'temp = 5' and 'i = 7' then there is no point of going forward since we already know it will not lead us to the 'targetSum'. Hence I am checking if vis[temp - 1][i] is not visited before going forward. 

I hope this will give you a clear picture of why I have used it in the code. If not, let me know, I will try to explain it differently and feel free to ask more question :-)
